### DISCUSSION TITLE
Subscribing to materialized views (like Hasura, RethinkDB and Firebase Realtime DB)

### DISCUSSION THREAD

**Discussion by jamieflorence** [supabase/supabase#3824] [Questions]
Howdy. I’ve been searching for a plausible Firebase alternative for a while now, and am delighted to have stumbled across this.

Before I get too excited, I have one quick question: is there any way to subscribe to a materialized view, rather than a table?

For context, I currently receive real time data which I put through a long computational pipeline to produce a summary table. The data comes in large packets at irregular intervals. I only care about the summary output in real time, not the individual data inputs, so I just want to be able to subscribe over websockets so I can build a dashboard.

Is this, or something like this, possible with supabase?

Hasura is able to, but their model uses live queries, and the computational model I have is far too intensive for that; I want to be able to trigger the refresh rate as data comes in.

**Comment by dshukertjr** [comment]
Hi @jamieflorence!

Currently, Supabase only supports subscribing to realtime updates of tables and tables only. As a workaround, have you considered creating and entering the computational results into another table instead of using a materialized view? That way, you will be able to listen to realtime updates without an issue!

**Reply by jamieflorence** [reply]
Thanks @dshukertjr, good to get confirmation.

I considered that, but materialised views have nice caching properties. Also, with a table I'd have to wipe and delete it, creating a bunch of delete notifications to the subscription. Ah well, I'll keep searching

**Reply by dshukertjr** [reply]
@jamieflorence Yeah, materialized views are pretty awesome, aren't they? I'm sure you have already considered various ways of making it work with a regular table, but could you some how minimize the number of deletes for every new summary data? Maybe instead of wiping the whole table every time, update the necessary data, and delete those rows there were not updated or something?

**Reply by baderdean** [reply]
@jamieflorence what alternative solution did you find in the meantime please?

**Comment by baderdean** [comment]
Hello,

is this feature in your roadmap please? And how about RLS checking also in Realtime?
Our goal is to use Realtime (Postgres Changes) as a way to stream data to the frontend of every user from at least 3 different tables.

regards

**Reply by dshukertjr** [reply]
>And how about RLS checking also in Realtime?

RLS is supported in Realtime now. 

>Our goal is to use Realtime (Postgres Changes) as a way to stream data to the frontend of every user from at least 3 different tables.

You can subscribe to those three tables and combine the data on the client side!

**Reply by baderdean** [reply]
Great news for RLS!
However the solution to rely on the client side presents several drawbacks:
- heavier load on the client side especially if there is a lot of data (our case something like 10 000 rows in front is usual)
- tightly coupling: expose too much of the internal backend data model to the frontend

For the moment we're going to create a dedicated table for the front end which is updated every X seconds from the other tables. At least we're not creating one for every user thanks to RLS support ^^

**Comment by bijanv** [comment]
Another request for this! We have some incredibly large views per user (100K rows) that would not be feasible to manage on the client side. Would love to see views have the ability to be subscribed to.

**Comment by toobulkeh** [comment]
+1 for a request, because it adds optionality of methods / patterns to de-normalize our db data.

**Comment by chreds** [comment]
One approach (I've been using) is with React Query, instead of worrying about having to edit displayed data you can refresh current data on changes. Would still require multiple subscribes for each source of the view, but at least you aren't having to deal with morphing specific underlying data based on updates.

**Comment by Norbiros** [comment]
Related to https://github.com/supabase/realtime-js/issues/185

**Comment by aishik-pyne** [comment]
Another vote for the feature request
Currenty I have multiple tables in a private schema and then a view in a public schema with RLS

I want to listen to the view so that changes in any of the tables in the private table can trigger the client to fetch the changes in the view
